src/CodeAnalysis/CSharp/MetaAnalyzers/CSharpImmutableObjectMethodAnalyzer.cs

```diff
@@ -0,0 +1,108 @@
+﻿using Microsoft.CodeAnalysis.Diagnostics;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Analyzers;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
```

Comment:
Sort the `using`s so the System ones come first.

Also, I think we need a copyright header.